### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by Silex** [Silex/docker.el#122]
https://github.com/Silex/docker.el/blob/master/docker-utils.el#L34

`shell-file-name` has to be `proxycmd.exe`

@OldIMP: do you see something else? do you have an idea about how to solve this elegantly?

**Comment by OldIMP** [comment]
Thx for the issue!

Actually this might be the only issue on windows, Right now I've just overwritten the function in my own .emacs as a workaround:
```elisp
(defun docker-utils-shell-command-to-string (command)
  "Overwrite original to avoid hardcoded /bin/sh
cf. https://github.com/Silex/docker.el/commit/5027a3d541b1dcbb2f7ec0bac04a30dceeca7ce8"
    (shell-command-to-string command))
```
and everything works fine.

Basically I think it's a good idea to let emacs decide `shell-file-name` by itself. I don't know what was exactly the problem w/ fish. Maybe I can help if u provide more details on that, since I've found no Github issues about the fish problem.

If the problem lies on the fish shell itself, maybe we can define a mode switch variable like `docker-shell-mode` as a suboptimal solution.

Thx again for the great job!!! I've been using `C-c d` on windows everyday and it worked perfectly!

**Comment by Silex** [comment]
The issues with fish were #66 and #121.

There was two issues:
1. fish was being used when shelling inside containers, and of course there it was not installed.
1. fish was being used instead of sh/bash when calling docker and it created parsing issues.

`1` was solved with `docker-container-shell-file-name`, and for `2` I explicitely set `shell-file-name` to sh to avoid parsing issues.

I think there's not many solution: we need to test for windows and set `shell-file-name` to `cmdproxy.exe` instead.

**Comment by Silex** [comment]
Ok it should be fixed by ed92726.

Please reopen if it still happens.
